src/Sentry/ViewHierarchy.cs

```diff
@@ -0,0 +1,57 @@
+using Sentry.Extensibility;
+using Sentry.Internal.Extensions;
+
+namespace Sentry
+{
+    public class ViewHierarchy : IJsonSerializable
+    {
+        public string RenderingSystem { get; set; } = string.Empty;
+        public List<IJsonSerializable>? Children { get; set; }
+
+        public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
+        {
+            writer.WriteStartObject();
+
+            writer.WriteStringIfNotWhiteSpace("rendering_system", RenderingSystem);
+
+            if (Children is {} children)
+            {
+                writer.WriteStartArray("windows");
+                foreach (var child in children)
+                {
+                    child.WriteTo(writer, logger);
+                }
+                writer.WriteEndArray();
+            }
+
+            writer.WriteEndObject();
+        }
+    }
+
+    public class ViewHierarchyNode : IJsonSerializable
```

Comment:
Wondering if `ViewHierarchytNode` is useful on its own, or if it will always need to be overridden with a more specialized class?  If the latter, then perhaps it should be `abstract`?